zerver/models.py

```diff
@@ -1500,3 +1548,43 @@ class RealmAuditLog(models.Model):
     event_type = models.CharField(max_length=40) # type: Text
     event_time = models.DateTimeField() # type: datetime.datetime
     backfilled = models.BooleanField(default=False) # type: bool
+
+class CustomProfileField(models.Model):
+    realm = models.ForeignKey(Realm)  # type: Realm
+    name = models.CharField(max_length=100)  # type: Text
+
+    NUMBER = 1
+    SHORT_TEXT = 2
+    LONG_TEXT = 3
+
+    FIELD_TYPE_CHOICES = (
+        (NUMBER, u'Number'),
+        (SHORT_TEXT, u'Short Text'),
+        (LONG_TEXT, u'Long Text'),
+    )
+    field_type = models.PositiveSmallIntegerField(choices=FIELD_TYPE_CHOICES,
+                                                  default=SHORT_TEXT)  # type: int
+
+    class Meta(object):
+        unique_together = ('realm', 'name')
+
+    def as_dict(self):
+        # type: () -> Dict[str, Union[int, Text]]
+        return {
+            'id': self.id,
+            'name': self.name,
+            'type': self.field_type,
+        }
+
+def custom_profile_fields_for_realm(realm_id):
+    # type: (int) -> List[Dict[str, Union[int, Text]]]
+    return Realm.objects.get(pk=realm_id).profile_fields
+
+class CustomProfileFieldValue(models.Model):
+    user_profile = models.ForeignKey(
+        UserProfile, related_name='custom_profile_field_values')  # type: UserProfile
+    custom_profile_field = models.ForeignKey(CustomProfileField)  # type: CustomProfileField
```

Comment:
the variable name here can probably be just `field` since it should be clear from context.